File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int>result;
        unordered_set<int>s(nums1.begin(),nums1.end());
        for(auto it:nums2){
            if(s.find(it)!=s.end()){
                result.push_back(it);
                s.erase(it);
            }
        }
        return result;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
         int n=nums1.length;
         int m=nums2.length;
        ArrayList<Integer>result=new ArrayList<>();
        HashSet<Integer>st = new HashSet<>();

        for(int i=0;i<n;i++){
            st.add(nums1[i]);
        }

        for(int i=0;i<m;i++){
            if(st.contains(nums2[i])){
                result.add(nums2[i]);
                st.remove(nums2[i]);
            }

        }
        int [] ans=new int[result.size()];
        for(int i=0;i<result.size();i++){
            ans[i]=result.get(i);
        }

        return ans;
    }
}